src/Microsoft.AspNetCore.Razor.Runtime/TagHelpers/TagHelperAttribute.cs

```diff
@@ -8,77 +8,70 @@ namespace Microsoft.AspNetCore.Razor.TagHelpers
     /// <summary>
     /// An HTML tag helper attribute.
     /// </summary>
-    public class TagHelperAttribute : IReadOnlyTagHelperAttribute
+    public class TagHelperAttribute
     {
         private static readonly int TypeHashCode = typeof(TagHelperAttribute).GetHashCode();
```

Comment:
Get rid of this and provide a `GetHashCode()` implementation using all property values. That's possible now that `TagHelperAttribute` is immutable and may improve the efficiency of `TagHelperAttributeList`.